lib/nerves_hub_link/features/geo/resolver.ex

```diff
@@ -0,0 +1,22 @@
+defmodule NervesHubLink.Features.Geo.Resolver do
+  @typedoc "Supported location sources"
+  @type sources() :: :gps | :geoip | :custom
```

Comment:
Since users can add their own this should probably be a string :)
(was this lifted from geo, if so, kind of irrelevant to the PR)

Or `{:custom, "where_i_put_it"}` style. Give us something to display in UI about how it gets it's data.